lib/shortener.rb

```diff
@@ -5,18 +5,25 @@ module Shortener
   autoload :ActiveRecordExtension, "shortener/active_record_extension"
   autoload :ShortenUrlInterceptor, "shortener/shorten_url_interceptor"
 
+  AMBIGUOUS_CHARACTERS = ['i', 'o', 'l', 'I', 'O', '0', '1'].freeze
+
   CHARSETS = {
-    alphanum: ('a'..'z').to_a + (0..9).to_a,
-    alphanumcase: ('a'..'z').to_a + ('A'..'Z').to_a + (0..9).to_a
+    alphanum: ('a'..'z').to_a + (0..9).map(&:to_s).to_a,
+    alphanumcase: ('a'..'z').to_a + ('A'..'Z').to_a + (0..9).map(&:to_s).to_a,
+    alphanum_no_ambiguous: ('a'..'z').to_a + (0..9).map(&:to_s).to_a - AMBIGUOUS_CHARACTERS,
+    alphanumcase_no_ambiguous:
+      ('a'..'z').to_a + ('A'..'Z').to_a + (0..9).map(&:to_s).to_a - AMBIGUOUS_CHARACTERS
   }
 
   # default key length: 5 characters
   mattr_accessor :unique_key_length
   self.unique_key_length = 5
 
   # character set to chose from:
-  #  :alphanum     - a-z0-9     -  has about 60 million possible combos
-  #  :alphanumcase - a-zA-Z0-9  -  has about 900 million possible combos
```

Comment:
Note that the number of possible combinations is dependent on both the character set and the unique key length. As the default key length is only a default, providing an (approximate) count seems inappropriate and incorrect.